### DISCUSSION TITLE
using with esp8266

### DISCUSSION THREAD

**Discussion by assegid** [Paciente8159/uCNC#768] [General]
Can someone please provide simplified instructions on how to use this library with the esp8266 board on a Mac and Arduino IDE? Will it work at all? I think I read somewhere that boards other than AVRs can only be used with windows?

**Comment by Paciente8159** [comment]
I there.
This is not a library (although it might be possible to convert it into one).

I recommend using VSCode and PlatformIO extension.

But you can build the code using Arduino.
Follow these [instructions](https://github.com/Paciente8159/uCNC/wiki/ESP8266:-Compiling) and install the ESP8266 boards in the board manager.
Select the LOLIN WEMOS D1 board
Open the uCNC.ino sketch
To use the configuration for the Wemos D1 board modify the [BOARD line](https://github.com/Paciente8159/uCNC/blob/0040f4edf5b8615809b251c1695cb799338f44c3/uCNC/cnc_config.h#L55C3-L55C4) to:

```
#ifndef BOARD
#define BOARD "esp8266/boardmap_wemos_d1.h"
#endif
```

Also comment the line in build_opt.h or simply delete that file (this is a file specific for STM32 builds but seems to cause a build error under Arduino)

You should be able to get the code compiled like this

**Reply by assegid** [reply]
I made the changes you suggested and the code did compile in Arduino. But where can I see the pin assignments for the Step and Dir lines? I assumed that assignment can be made using the boardmap_overrides.h file that can be generated using the web tool. But when I did that compilation fails with the error on the #include "boardmap_reset.h" line. Is it ok to just comment out that line?

**Reply by Paciente8159** [reply]
Also note that the default pins for that board are defined in [this file](https://github.com/Paciente8159/uCNC/blob/master/uCNC/src/hal/boards/esp8266/boardmap_wemos_d1.h)

**Comment by Paciente8159** [comment]
Did you add the files included in the zip to the project?
That file does not exist in the default project. It's generated by the web builder and must be added.

**Reply by assegid** [reply]
Thanks a lot. That made it work. i also found the default pin assignments following the link you provided. Looking forward to testing the setup with UGS!

**Reply by Paciente8159** [reply]
If you get problems connecting to UGS change [this value](https://github.com/Paciente8159/uCNC/blob/785c01400f4a92327e11af8bff288658348312ac/uCNC/cnc_config.h#L606) to 2

**Reply by assegid** [reply]
So UGS connects but it is permanently in Alarm state. I am trying to test it with a simple setup with 2 motors using step and dir lines only. No limit switches and the !Sleep pin of the stepper driver is held permanently high to control the motors with only step and dir. I was hoping to test them with the Jog feature after a reset, but the system displays alarm and stays there forever. The response of the esp8266 as soon as it is connected to UGS is below.

`*** Fetching device settings
>>> $$
$0 = 4294967665.2147483887    (Step pulse time, microseconds)
$2 = 255    (Step pulse invert, mask)
$3 = 255    (Step direction invert, mask)
$4 = 255    (Invert step enable pin, boolean)
$5 = 255    (Invert limit pins, boolean)
$6 = 255    (Invert probe pin, boolean)
$7 = 255   
$10 = 255    (Status report options, mask)
$11 = 4294967665.2147483887    (Junction deviation, millimeters)
$12 = 4294967665.2147483887    (Arc tolerance, millimeters)
$13 = 255    (Report in inches, boolean)
$20 = 255    (Soft limits enable, boolean)
$21 = 255    (Hard limits enable, boolean)
$22 = 255    (Homing cycle enable, boolean)
$23 = 65535    (Homing direction invert, mask)
$24 = 65535    (Homing locate feed rate, mm/min)
$25 = 65535    (Homing search seek rate, mm/min)
$26 = 65535    (Homing switch debounce delay, milliseconds)
$27 = 4294967665.2147483887    (Homing switch pull-off distance, millimeters)
$30 = 65535    (Maximum spindle speed, RPM)
$31 = 65535    (Minimum spindle speed, RPM)
$32 = 255    (Laser-mode enable, boolean)
$81 = 4294967665.2147483887   
$100 = 4294967665.2147483887    (X-axis travel resolution, step/mm)
$101 = 4294967665.2147483887    (Y-axis travel resolution, step/mm)
$110 = 4294967665.2147483887    (X-axis maximum rate, mm/min)
$111 = 4294967665.2147483887    (Y-axis maximum rate, mm/min)
$120 = 4294967665.2147483887    (X-axis acceleration, mm/sec^2)
$121 = 4294967665.2147483887    (Y-axis acceleration, mm/sec^2)
$130 = 4294967665.2147483887    (X-axis maximum travel, millimeters)
$131 = 4294967665.2147483887    (Y-axis maximum travel, millimeters)
ok`

My boardmap_override.h file is below...
`#ifndef BOADMAP_OVERRIDES_H
#define BOADMAP_OVERRIDES_H
#ifdef __cplusplus
extern "C"
{
#endif

#include "boardmap_reset.h"
#define MCU MCU_ESP8266
#define KINEMATIC KINEMATIC_CARTESIAN
#define AXIS_COUNT 2
#define TOOL_COUNT 1
#define BAUDRATE 115200
#define BOARD_NAME "oz8266WEMOS D1"
#define UART_PORT 0
#define STEP0_BIT 12
#define STEP0_PORT D
#define STEP1_BIT 15
#define STEP1_PORT D
#define STEP2_PORT D
#define DIR0_BIT 14
#define DIR0_PORT D
#define DIR1_BIT 13
#define DIR1_PORT D
#define DIR2_PORT D
#define STEP0_EN_BIT 0
#define STEP0_EN_PORT D
#define PWM0_PORT D
#define DOUT0_BIT 16
#define DOUT0_PORT D
#define TX_BIT 1
#define TX_PORT D
#define RX_BIT 3
#define RX_PORT D
//Custom configurations
#define RX_PULLUP 
#define RX_PULLUP 


#ifdef __cplusplus
}
#endif
#endif
`
What I am I missing???

**Reply by Paciente8159** [reply]
Is this the latest version? If yes the settings need to be propelly reset the first time. Send a `$RST=*` command.
I'll be able to test your settings tomorrow and give test it for my self

**Reply by Paciente8159** [reply]
Also there is something odd with settings values. Try that reset command to see if it fixes it.

**Reply by assegid** [reply]
I tried reset and unlock many, many times. It goes into Alarm immediately.

**Reply by Paciente8159** [reply]
Tomorrow I will test on my board and give you feedback

**Reply by assegid** [reply]
Looking forward to what you will find out. BTW, I don't think it will make any difference, but I did not send the reset from the console. I clicked on the reset and unlock buttons of the user interface of UGS. I will also try the console, just in case tomorrow. I am currently away from the board.

**Reply by Paciente8159** [reply]
I've tested both the default configuration and a configuration based on the pins you sent and all is working as expected on my WeMos D1 R2 and using VSCode and PlatformIO. Arduino IDE is a bit more problematic but I was able to compile and upload by disabling the wifi option.

I don't know how you added the files to the project but here is the project configuration I used
[uCNC.zip](https://github.com/user-attachments/files/17554739/uCNC.zip)

**Reply by assegid** [reply]
I programmed the 8266 board with project configuration, but getting the same results. I have attached screen capture of UGS and the wiring schematic. Do you see any issues?
<img width="1258" alt="ugsState" src="https://github.com/user-attachments/assets/074abcc0-2a22-42dc-9103-d6ebdd1648c9">
<img width="1258" alt="ugsState" src="https://github.com/user-attachments/assets/b231a3a9-4a42-4d90-a9d8-1280f27afb30">
<img width="825" alt="schematic" src="https://github.com/user-attachments/assets/7af93506-6025-42f1-bb28-e094baee518e">

**Reply by Paciente8159** [reply]
What are you using to program the board? Arduino IDE or VSCode + PIO?

**Reply by assegid** [reply]
Arduino IDE.

**Reply by Paciente8159** [reply]
Can you give a try with VSCode and PlatformIO?

**Reply by Paciente8159** [reply]
On Arduino IDE which board did you selected?
These are my settings
![{DEED2DB4-554E-499B-A1E8-DDEE3C429C72}](https://github.com/user-attachments/assets/ac674814-c6aa-494b-80f0-fe5c7ce442a9)

**Reply by assegid** [reply]
I selected Lolin(Wemos) D1 mini Pro. There were no complaints during programming, but I will match your settings and try. It will take me a while setting up PIO, but I am going to try it also.
<img width="499" alt="arduino" src="https://github.com/user-attachments/assets/26379b0e-e513-424d-aa1a-a36db6a0200f">

**Reply by Paciente8159** [reply]
The Flash size difference should explain the odd settings values.
This latest version of µCNC prevents any motion or action with invalid settings loaded that is why it was always in locked/Alarm mode

**Reply by assegid** [reply]
i think that difference is because I am using the D1 mini pro which has 16MB flash. The only way to change that is by selecting D1 R2 mini, which I did just for testing purposes. I was actually surprised that the IDE allowed me to upload the program by selecting D1 R2 instead of D1 pro, but it did. Nevertheless, the result was the same; stuck in Alarm state.

**Reply by Paciente8159** [reply]
Then you can choose your board again.
Try uncommenting this setting
https://github.com/Paciente8159/uCNC/blob/785c01400f4a92327e11af8bff288658348312ac/uCNC/cnc_config.h#L128

Remeber you will loose all your settings each power up/down

**Reply by assegid** [reply]
That did not work, but uncommenting lines 388, 574, 577 and 580 did it. Thank you so much for your help!!!

**Reply by Paciente8159** [reply]
Ok. I bet line 574 did the trick.
By the way did you changed this line?
https://github.com/Paciente8159/uCNC/blob/785c01400f4a92327e11af8bff288658348312ac/uCNC/cnc_config.h#L55

If not you are probably loading arduino uno pinout under your overrides.
Later I'll check if that is the problem

**Comment by assegid** [comment]
"with **your** project configuration"
